### PR TITLE
ABR-40 Ceremonial Rifle

### PR DESCRIPTION
**Pull request by PhantomEpicness**
## About The Pull Request
Adds the ABR-40 ceremonial rifle, a sleek polished rifle based on the L42A meant for ceremonies, drills, and execution. Credit to Blithering for the sprites

> An ABR-40 battle rifle, the L42A's predecessor, with an authentic polished wood furniture for ceremonial usage. Its frame is shined to perfection and you can even see your reflection in the bolt. Lacks the threaded barrel and underbarrel rail for attachments, with only the bayonet lug being the only place to attach anything. Too cumbersome to be fired with one hand, and the Sergeant Major would kill any marine caught trying.

Statistically, it sports a slightly faster fire rate, though it can only mount a bayonet, has worse wield delay, spawns with a half magazine, and movement slowdown due to a non-detachable stock, making it more of a side grade to the L42. It also cannot be fired while one-handed as it is shouldered while doing so.

The rifles can be found in the new room, execution chamber (replaced l42s there), and the SEA gear vendor. They have their own "ABR-40" magazines, which are essentially just half-size L42A mags, can be found in execution and purchasable in SEA vendor. L42 is compatible with ABR-40 mags and visa versa.

image
image
image
image
image

I also added M41A and ABR-40 blank magazines, which can be found in the SEA vendor and the new ceremonial room near the memorial. 
image

Two boxes of blank magazines are in the ceremonial room and blank magazines can be ordered in SEA gear vendor. Additionally, a box can also be found in maint

In addition, this PR also adds the ABR-40, which is in code only as of right now. ABR-40 is L42,but with worse fire delay and wield delay.

Also gave the SEA uniform vendor service jackets and dress shoes as a request from a mentor.

Both blank and ABR-40 magazines are buyable at cheap prices at ASRS

## Why It's Good For The Game
soul

## Changelog

:cl: Totalepicess, Blithering
add: Added M41A and L42A blank ammo
add: Added the ABR-40 Ceremonial Rifle, found in SEA vendor, execution and ceremonial room. Sprites by Blithering
add: Added a new ceremonial room to the Almayer, next to the memorial. 
add: Added service jacket to SEA vendor
/:cl:

### CODE REVIEW

**Comment by Superwelder** [comment]
We ROTC now kids! Next up, SEAs get to administer the ASVAB to new players!

**Comment by FoxyStalin** [comment]
Your changelog is default.

**Comment by RadiationFan** [comment]
Could you add this to PFC for 45 or 30 points for the sovl

**Review by Agameofscones — APPROVED** [review]
sprites look good

**Review by nauticall — APPROVED** [review]
Sprites look good to me too, and mapping's good. Map approved toooo

though I will say I was working on an M1 Garand, ya beat me to a proper ceremonial rifle x.x

**Comment by PhantomEpicness** [comment]
> Could you add this to PFC for 45 or 30 points for the sovl

Err no,

Go steal it from ceremonial room/ask an MP or SEA

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -903,6 +903,18 @@
 	. = ..()
 	M.AddComponent(/datum/component/bonus_damage_stack, 10, world.time)
 
+/datum/ammo/bullet/rifle/blank
+	name = "blank rifle bullet"
+	icon_state = "shrapnel_bright2"
```
Comment (stanalbatross) [inline code/modules/projectiles/ammo_datums.dm:908]: we have a dedicated blank bullet icon iirc

File: code/modules/projectiles/guns/rifles.dm [hunk 2]
```diff
@@ -1243,10 +1243,60 @@
...
+/obj/item/weapon/gun/rifle/l42a/abr40/drill/set_gun_config_values()
+	..()
+	fire_delay = FIRE_DELAY_TIER_9 // Polished parts
+
+/obj/item/weapon/gun/rifle/l42a/abr40
+	name = "\improper ABR-40 rifle"
+	desc = "Essentially the civilian predessessor to the L42A, the ABR-40 bears similarities to the L42 in inner workings while sporting a 'hunting rifle' esque wood colored furniture. Legal in California!"
+	icon_state = "l42wood"
+	item_state = "l42wood"
+	current_mag = /obj/item/ammo_magazine/rifle/l42a
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/rifles.dm:1273]: should it not start with a half mag

File: code/modules/projectiles/guns/rifles.dm [hunk 3]
```diff
@@ -1243,10 +1243,60 @@
...
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/bayonet/upp,
+						/obj/item/attachable/bayonet/c02
+						)
+	wield_delay = WIELD_DELAY_FAST
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_WIELDED_FIRING_ONLY
+	starting_attachment_types = list(/obj/item/attachable/stock/carbine/wood/drill)
+
+/obj/item/weapon/gun/rifle/l42a/abr40/drill/set_gun_config_values()
+	..()
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/rifles.dm:1265]: subtype goes BELOW parent

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/datums/elements/bullet_trait/del_range.dm [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/*
```
[the remark this answers is not in the document]
Comment (PhantomEpicness) [reply code/datums/elements/bullet_trait/del_range.dm:1]: leftover from testing
removed

**Comment by PhantomEpicness** [comment]
Ready for review

Changes:

Blank magazines and ABR-40 mag box now buyable in req

File: code/modules/projectiles/ammo_boxes.dm [hunk 5]
```diff
@@ -454,6 +469,25 @@
...
 
+/obj/item/ammo_box/magazine/l42a/half
+	name = "magazine box (HALF-SIZE L42A x 32)"
+	num_of_magazines = 32
+	magazine_type = /obj/item/ammo_magazine/rifle/l42a/half
+
+/obj/item/ammo_box/magazine/l42a/half/empty
+	empty = TRUE
+
+/obj/item/ammo_box/magazine/l42a/half/blank
```
Comment (morrowwolf) [inline code/modules/projectiles/ammo_boxes.dm:480]: Magazines inside the box seem to not appear correctly when deployed.

File: code/modules/projectiles/ammo_boxes.dm [hunk 6]
```diff
@@ -352,6 +352,21 @@
 /obj/item/ammo_box/magazine/explosive/empty
 	empty = TRUE
 
+/obj/item/ammo_box/magazine/blank
```
Comment (morrowwolf) [inline code/modules/projectiles/ammo_boxes.dm:355]: Magazines inside the box seem to not appear correctly when deployed.

**Review by morrowwolf — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/magazines/rifles.dm [hunk 7]
```diff
@@ -290,6 +296,21 @@
...
+	icon_state = "l42mk1_half"
+	gun_type = /obj/item/weapon/gun/rifle/l42a
+	max_rounds = 12
+
+
+/obj/item/ammo_magazine/rifle/l42a/half/blank
+	name = "\improper L42A blank magazine (10x24mm)"
+	desc = "A magazine loaded with blank rounds intended for the ceremonies, most notably being the centuries old tradition of the 21 gun salute. Despite a clear white strip to prevent confusion with normal rounds, there is always that one private that somehow manages to bring these to the frontline every now and then."
+	icon_state = "l42mk1_blank"
+	gun_type = /obj/item/weapon/gun/rifle/l42a
```
Comment (morrowwolf) [inline code/modules/projectiles/magazines/rifles.dm:312]: gun_type set by parent

**Review by morrowwolf — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammo_boxes.dm (hunk 6 above)
Comment (PhantomEpicness) [reply code/modules/projectiles/ammo_boxes.dm:355]: Fixed

**Comment by PhantomEpicness** [comment]
Fixed

File: code/modules/projectiles/ammo_datums.dm [hunk 8]
```diff
@@ -903,6 +903,18 @@
 	. = ..()
 	M.AddComponent(/datum/component/bonus_damage_stack, 10, world.time)
 
+/datum/ammo/bullet/rifle/blank
+	name = "blank rifle bullet"
+	icon_state = "blank"
+	damage = 50  //Full power blanks that create big muzzleflash can fuck you up.
```
Comment (morrowwolf) [inline code/modules/projectiles/ammo_datums.dm:909]: Drop this to 10.

File: code/modules/projectiles/gun_attachables.dm [hunk 9]
```diff
@@ -1407,6 +1407,25 @@ Defined in conflicts.dm of the #defines folder.
 	recoil_unwielded_mod = RECOIL_AMOUNT_TIER_4
 	scatter_unwielded_mod = SCATTER_AMOUNT_TIER_8
 
+/obj/item/attachable/stock/carbine/wood
+	name = "\improper L42 wooden stock"
+	desc = "A fully polished wood stock, heavy yet sturdy. Attaches to the L42A Battle Rifle. You can bash someone good with this."
```
Comment (morrowwolf) [inline code/modules/projectiles/gun_attachables.dm:1412]: Make explicit that the wood is fake.

File: code/modules/projectiles/gun_attachables.dm [hunk 10]
```diff
@@ -1407,6 +1407,25 @@ Defined in conflicts.dm of the #defines folder.
...
+
+/obj/item/attachable/stock/carbine/wood/Initialize()
+	..()
+	accuracy_mod = HIT_ACCURACY_MULT_TIER_8
+	recoil_mod = -RECOIL_AMOUNT_TIER_3
+	scatter_mod = -SCATTER_AMOUNT_TIER_9
+
+/obj/item/attachable/stock/carbine/wood/drill
+	name = "\improper L42 exotic wood stock"
+	desc = "A fully polished non detachable wood stock attached to select L42A ceremonial rifles. You can bash someone good with this."
```
Comment (morrowwolf) [inline code/modules/projectiles/gun_attachables.dm:1426]: Missed it before but this is the same desc as parent

File: code/modules/projectiles/guns/rifles.dm [hunk 11]
```diff
@@ -1243,10 +1243,58 @@
 	damage_falloff_mult = 0
 	scatter = SCATTER_AMOUNT_TIER_8
 
-
 /obj/item/weapon/gun/rifle/l42a/training
 	current_mag = /obj/item/ammo_magazine/rifle/l42a/rubber
 
+/obj/item/weapon/gun/rifle/l42a/abr40
+	name = "\improper ABR-40 rifle"
+	desc = "Essentially the civilian predecessor to the L42A, the ABR-40 bears similarities to the L42 in inner workings while sporting a 'hunting rifle' esque wood colored furniture. Legal in California!"
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/rifles.dm:1251]: Make explicit that it's fake wood and remove reference to California legality.

File: code/modules/projectiles/guns/rifles.dm [hunk 12]
```diff
@@ -1243,10 +1243,58 @@
...
+
+/obj/item/weapon/gun/rifle/l42a/abr40/drill/set_gun_config_values()
+	..()
+	fire_delay = FIRE_DELAY_TIER_7
+
+
+// Essentially L42 Custom, though more of a sidegrade due to 0 attachments, higher weild delay and the fact that you have to find better magazines.
+/obj/item/weapon/gun/rifle/l42a/abr40/drill
+	name = "\improper ABR-40 Ceremonial Rifle"
+	desc = "An ABR-40 battle rifle, the L42A's predecessor, with an authentic polished wood furniture for ceremonial usage. Its frame is shined to perfection and you can even see your reflection in the bolt. Lacks the threaded barrel and underbarrel rail for attachments, with only the bayonet lug being the only place to attach anything. Too cumbersome to be fired with one hand, and the Sergeant Major would kill any marine caught trying."
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/rifles.dm:1285]: Make explicit wood is fake.

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Lore team had some concerns.

File: code/modules/projectiles/gun_attachables.dm (hunk 10 above)
Comment (PhantomEpicness) [reply code/modules/projectiles/gun_attachables.dm:1426]: diff desc this time ( if people can somehow see it)

File: code/modules/projectiles/guns/rifles.dm (hunk 12 above)
Comment (PhantomEpicness) [reply code/modules/projectiles/guns/rifles.dm:1285]: replica wood

(:

**Review by morrowwolf — APPROVED** [review]
Sammy forgot to mark mapping approved so I've done so.

Code LGTM

I think lore team concerns have been addressed.

Some ML concerns were brought up so I'll leave this to Stan to merge.

File: code/modules/projectiles/ammo_boxes.dm [hunk 13]
```diff
@@ -352,6 +353,22 @@
...
+	overlay_content = "_training"
+	overlay_gun_type = "_m41"
+	magazine_type = /obj/item/ammo_magazine/rifle/blank
+
+/obj/item/ammo_box/magazine/blank/empty
+	empty = TRUE
+
+/obj/item/ammo_box/magazine/blank/sus
+	name = "odd magazine box (WP M41A x 10)"
+	desc = "An old beat up ammuniton. A badly written message in white crayon reads 'BRAND NEW M41 WP ROUNDS'"
```
Comment (stanalbatross) [inline code/modules/projectiles/ammo_boxes.dm:370]:
```suggestion
```

File: code/modules/projectiles/guns/rifles.dm [hunk 14]
```diff
@@ -1243,10 +1243,58 @@
...
+	desc = "An ABR-40 battle rifle, the L42A's predecessor, with a fully polished and almost indistinguishable replica-wood furniture for ceremonial usage. Its frame is shined to perfection and you can even see your reflection in the bolt. Lacks the threaded barrel and underbarrel rail for attachments, with only the bayonet lug being the only place to attach anything. Too cumbersome to be fired with one hand, and the Sergeant Major would kill any marine caught trying."
+	attachable_allowed = list(
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/bayonet/upp,
+						/obj/item/attachable/bayonet/c02
+						)
+
+/obj/item/weapon/gun/rifle/l42a/abr40/drill/set_gun_config_values()
+	..()
+	fire_delay = FIRE_DELAY_TIER_9 // Polished parts
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/rifles.dm:1294]: /drill being procced twice

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/ammo_boxes.dm (hunk 13 above)
Comment (PhantomEpicness) [reply code/modules/projectiles/ammo_boxes.dm:370]: instanced instead

**Comment by PhantomEpicness** [comment]
ready for review

**Comment by PhantomEpicness** [comment]
no clue why the map diffbot is going nuts but everything else is checking out fine

**Comment by Agameofscones** [comment]
resolve conflict

**Comment by PhantomEpicness** [comment]
> resolve conflict

done

**Comment by kiVts** [comment]
trolled

**Comment by PhantomEpicness** [comment]
> trolled

r4t moment

fixed

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**